src/dra/5D6C4.c

```diff
@@ -474,7 +474,160 @@ void AddHearts(s32 value) {
     }
 }
 
-INCLUDE_ASM("dra/nonmatchings/5D6C4", func_800FE97C);
+// Note: Arg3 is unused, but is given in the call from func_80113D7C
+s32 func_800FE97C(Unkstruct_800FE97C* arg0, s32 arg1, s32 arg2, s32 arg3) {
+    s32 ret;
+    s32 var_s1;
+    s32 itemCount;
+
+    var_s1 = arg2;
```

Comment:
you can remove var_s1 by replacing it entirely by arg2
https://decomp.me/scratch/Qyv5F